scripts/TPI.py

```diff
@@ -0,0 +1,74 @@
+import numpy as np
+import scipy.optimize as opt
+
+
```

Comment:
Define `TPI` as a function so that we can call this in `execute.py`. Just the same as what we did in class